src/oxr/index.ts

```diff
@@ -0,0 +1,60 @@
+export type OXEClient = ReturnType<typeof createOpenExchangeClient>
+
+export type OXEInternationalCurrencies = "USD" | "EUR" | "GBP" | "JPY" | "BRL";
+
+export interface OXEQueryParams {
+  'app_id': string,
+  readonly [key: string]: string | number | boolean;
+}
+
+export interface OXEDollarRates {
+  rates: Record<string, number>;
+  base: string;
+  timestamp: number;
+  license: string;
+  disclaimer: string;
+}
+
+const request = (method: 'GET', path: string, query: OXEQueryParams): Promise<Response> => {
+  const url = `https://openexchangerates.org/api` + path + Object.entries(query).reduce((str, [key, value]) => str + `${str.length ? '&' : '?'}${key}=${value}`, '')
```

Comment:
I would move this reduce to a getQuery function:
```suggestion
  const getQuery = (query: OXEQueryParams) => Object.entries(query).reduce((str, [key, value]) => str + `${str.length ? '&' : '?'}${key}=${value}`, '');
  // [...]
  const url = `https://openexchangerates.org/api${path}${getQuery(query)}`
```
WDYT?

Reply:
I think this doesn't make much of a difference...

Reply:
It's just to improve readability.
Also, we can reuse the the getQuery function to build params string on further implementations.
That said, I don't really have problems with the way it is right now.